File: project1/lower.R
#!/usr/bin/env Rscript

process <- function (input) {
  # Do something more here!
  output <- tolower(input)
  return(output)
}


f <- file("stdin")
open(f)
while(length(line <- readLines(f, n=1)) > 0) {
  line = process(line)
  write(line, stdout())
}

File: project1/lower.py
#!/usr/bin/env python

"""
A filter that _________.
"""

import fileinput


def process(line):
    """For each line of input, _____."""
    line = line[:-1]
    # Do something more here!
    print(line.lower())


for line in fileinput.input():
    process(line)
